src/main/java/unicash/model/transaction/predicates/TransactionAmountContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,58 @@
+package unicash.model.transaction.predicates;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import unicash.commons.util.StringUtil;
+import unicash.commons.util.ToStringBuilder;
+import unicash.model.commons.Amount;
+import unicash.model.transaction.Transaction;
+
+/**
+ * Tests that a {@code Transactions}'s {@code Amount} matches any of the keywords given.
+ * Keywords in this context refers to amount strings, and given that the String value of
+ * any valid amount will always be a single word string, the entire amount has to match.
+ * This is unlike most of the other property predicates that only require a partial word
+ * match - the reason being, it does not make sense to match partial amounts in a find context.
+ *
+ * </p> Additionally, given the nature of the Amount String i.e. it containing the currency
+ * symbol, this predicate forms a String of the raw amount value directly, so keyword comparisons
+ * are valid regardless of currency used.
+ */
+public class TransactionAmountContainsKeywordsPredicate
+        implements Predicate<Transaction> {
+    private final List<String> keywords;
+
+    public TransactionAmountContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Transaction transaction) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(
+                        Amount.amountToDecimalString(transaction.getAmount()),
+                        keyword));
```

Comment:
I'm not too sure what you mean but I think maybe you're thinking of the `toString()` method, which prefixes the currency symbol. `amountToDecimalString` is actually the opposite, it simply returns the number itself `xx.xx` so when a user types `find 45.70`, transactions with `45.71` won't show up because of the more strict `containsWord` requirement. Using `Amount.toString()` together with `containsWord` wouldn't work as `$45.00` (the toString return) isn't considered the same word as `45.00` (the user input keyword), and using `containsSubstring` here wouldn't be appropriate, as every number would be matched (i.e. keyword `70` would match `45.70`)

`amountToDecimalString` was the first such method used to round an amount value and so this was used everywhere that a 2 dp precision would have been required as a raw string without a dollar symbol infront, back when amounts were not restricted to the current constraints. However, now that the `amountString` exists, and all amounts are constrained by default at creation time, I could totally switch this to `amountString` and it would work the same.

On that note, whoever is editing Amount might want to consider this duplicated method.